core/ecschema-editing/src/Validation/SchemaComparer.ts

```diff
@@ -838,4 +859,49 @@ export class SchemaComparer {
 
     await Promise.all(promises);
   }
+
+  /**
+   * Compares two item keys.
+   * @param itemKeyA item key A to compare to.
+   * @param itemKeyB item key B to compare to.
+   * @param topLevelSchemaNameA top level schema name in which the item A exists.
+   * @param topLevelSchemaNameB top level schema name in which the item B exists.
+   * @returns true if both names are the same and neither one has a referenced schema.
+   */
+  private areItemsSameByName(
+    itemKeyA: Readonly<SchemaItemKey> | undefined,
+    itemKeyB: Readonly<SchemaItemKey> | undefined,
+    topLevelSchemaNameA: string,
+    topLevelSchemaNameB: string | undefined ): boolean {
+
+    const nameA = itemKeyA ? itemKeyA.name : undefined;
+    const nameB = itemKeyB ? itemKeyB.name : undefined;
+
+    const schemaNameA = itemKeyA ? itemKeyA.schemaName : undefined;
+    const schemaNameB = itemKeyB ? itemKeyB.schemaName : undefined;
+
+    return nameA === nameB && schemaNameA === topLevelSchemaNameA && schemaNameB === topLevelSchemaNameB;
+  }
+
+  /**
+   * Looks for same classA in containerB using fullName first and if not found try using the keys.
+   * @param classNameA name of the class to look for in containerB.
+   * @param containerA container which classNameA belongs to.
+   * @param containerB container in which to look for classNameA.
+   * @returns true if a same classA is in containerB, otherwise false.
+   */
+  private containerHasClass(classNameA: string, containerA: CustomAttributeContainerProps, containerB: CustomAttributeContainerProps): boolean {
+    if (containerB && containerB.customAttributes) {
+      for (const caB of containerB.customAttributes) {
+        const classNameB = caB[0];
+        const classItemKeyA = containerA.schema.getSchemaItemKey(classNameA);
+        const classItemKeyB = containerB.schema.getSchemaItemKey(classNameB);
+        const areSameByName = this.areItemsSameByName(classItemKeyA, classItemKeyB, containerA.schema.name, containerB.schema.name);
+        if (classNameA === classNameB || areSameByName) {
```

Comment:
do you need to compare the full class names directly?  Shouldn't that case be covered by areItemsSameByName

Reply:
Looking at it closely and stepping through it, areItemsSameByName can cover a full name comparison. I think I could even take it a step further and remove fullName comparisons for other cases where areItemsSameByName is used like in baseClass comparisons.